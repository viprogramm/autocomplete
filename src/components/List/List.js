import React from "react";

import { func, array } from "prop-types";

const List = props => {
  const { items, onClick, className = "", itemRender: ItemRender } = props;

  return (
    <div className={className}>
      {items.map((item, index) => {
        return (
          <div key={item.id || index} onClick={onClick(item)}>
            {typeof item === "object" ? (
              typeof ItemRender === "function" ? (
                <ItemRender {...item} />
              ) : (
                item.name
              )
            ) : typeof ItemRender === "function" ? (
              <ItemRender item={item} />
            ) : (
              item
            )}
          </div>
        );
      })}
    </div>
  );
};

List.propTypes = {
  items: array,
  onClick: func
};

List.defaultProps = {
  items: [],
  onClick: () => () => {}
};

export default List;

{
  /*<div className={className}>*/
}
{
  /*{items.map((item, index) => {*/
}
{
  /*return (*/
}
{
  /*<div key={item.id || index} onClick={onClick(item)}>*/
}
{
  /*{typeof ItemRender === "function" ? (*/
}
{
  /*<ItemRender {...item} />*/
}
{
  /*) : item.name ? (*/
}
{
  /*item.name*/
}
{
  /*) : (*/
}
{
  /*item*/
}
{
  /*)}*/
}
{
  /*</div>*/
}
{
  /*);*/
}
{
  /*})}*/
}
{
  /*</div>*/
}
