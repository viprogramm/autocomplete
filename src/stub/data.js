const data = [
  { id: 1, name: "ETH", value: 20 },
  { id: 2, name: "ETC", value: 2.4 },
  { id: 3, name: "BTC", value: 1.05 }
];

export default data;
