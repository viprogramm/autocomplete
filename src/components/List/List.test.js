import React from "react";
import renderer from "react-test-renderer";
import List from "./List";
import data from "../../stub/data";

describe("List", () => {
  test("without any props", () => {
    const tree = renderer.create(<List />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("with simple items and active class name", () => {
    const items = data.map(item => item.name);
    const tree = renderer
      .create(<List className="active" items={items} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("with simple items and item renderer", () => {
    const items = data.map(item => item.name);

    const itemRender = item => {
      return `name: ${item}`;
    };

    const tree = renderer
      .create(<List className="active" items={items} itemRender={itemRender} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("with custom item renderer", () => {
    const itemRender = ({ name, value }) => {
      return `${name} (ballance: ${value})`;
    };
    const tree = renderer
      .create(<List items={data} itemRender={itemRender} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
