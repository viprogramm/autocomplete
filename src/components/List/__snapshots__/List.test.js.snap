// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List with custom item renderer 1`] = `
<div
  className=""
>
  <div
    onClick={[Function]}
  >
    ETH (ballance: 20)
  </div>
  <div
    onClick={[Function]}
  >
    ETC (ballance: 2.4)
  </div>
  <div
    onClick={[Function]}
  >
    BTC (ballance: 1.05)
  </div>
</div>
`;

exports[`List with simple items and active class name 1`] = `
<div
  className="active"
>
  <div
    onClick={[Function]}
  >
    ETH
  </div>
  <div
    onClick={[Function]}
  >
    ETC
  </div>
  <div
    onClick={[Function]}
  >
    BTC
  </div>
</div>
`;

exports[`List with simple items and item renderer 1`] = `
<div
  className="active"
>
  <div
    onClick={[Function]}
  >
    name: [object Object]
  </div>
  <div
    onClick={[Function]}
  >
    name: [object Object]
  </div>
  <div
    onClick={[Function]}
  >
    name: [object Object]
  </div>
</div>
`;

exports[`List without any props 1`] = `
<div
  className=""
/>
`;
